src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -374,6 +387,10 @@ static inline int MPIDI_OFI_do_am_isend(int rank,
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_OFI_DO_AM_ISEND);
     return mpi_errno;
   fn_fail:
+    if (data_sz && attr.type == MPL_GPU_POINTER_DEV) {
```

Comment:
I think to avoid double condition check here, we can define a bool flag for `host_alloc` around line 357. I did same for OFI fallback.